src/components/MyHeader/my-header.tsx

```diff
@@ -0,0 +1,46 @@
+import React, { useState }from 'react'
+import classes from './my-header.module.css'
+import { MyInput } from '../MyInput/my-input'
+import { useNavigate, useLocation } from 'react-router-dom';
+
+export function MyHeader(props:any) {
+    const [value, setValue] = useState(localStorage.getItem('search'))
+    const navigate = useNavigate();
+    const location = useLocation();
+    function checkingPosts() {
+        if(location.pathname === '/posts'){
+            navigate('/about');
+        }
+    }
+    function checkingAbout() {
+        if(location.pathname === '/about'){
+            navigate('/posts');
+        };
+    }
+
+    return (
+    <header role="header" className={classes.header}>
+        <div  className={classes.logo_container}>
+            <img className={classes.img} src='iStar_Design_Thanksgiving_LineIcons_Live-47-512.webp'></img>
+            <h1 className={classes.logo_top}>BIGPIE</h1>
+        </div>
+        <MyInput 
+        onBlur={(e:any) => {
```

Comment:
We were supposed to use class components to access lifecycle events, such as `componentDidMount` and `componentWillUnmount`. Restoring and saving input's value should be implemented inside of lifecycle events.